shared/src/business/useCases/serveCaseToIrs/serveCaseToIrsInteractor.js

```diff
@@ -162,10 +168,39 @@ const generateNoticeOfReceipt = async ({ applicationContext, caseEntity }) => {
   const caseConfirmationPdfName =
     caseEntity.getCaseConfirmationGeneratedPdfFileName();
 
+  const clinicLetterKey = getClinicLetterKey({
+    procedureType,
+    trialLocation: preferredTrialCity,
+  });
+
+  const doesClinicLetterExist = await applicationContext
+    .getPersistenceGateway()
+    .isFileExists({
+      applicationContext,
+      key: clinicLetterKey,
+    });
+
+  if (doesClinicLetterExist) {
+    const clinicLetter = await applicationContext
+      .getPersistenceGateway()
+      .getDocument({
+        applicationContext,
+        key: clinicLetterKey,
+        protocol: 'S3',
```

Comment:
Possible DOD, extract all of these s3 strings to an entity constant?

Reply:
Ooh that is a good idea